### PR TITLE
Code refactor in all projects

### PR DESCRIPTION
**Pull request by GermanAizek**
More info: https://cwe.mitre.org/data/definitions/190.html

### CODE REVIEW

**Comment by GermanAizek** [comment]
@derekmauro review my pull request and report inaccuracies

**Review by derekmauro — CHANGES_REQUESTED** [review]
Please don't fix bugs and arbitrary style issues in the same pull request. Please create a separate pull request for anything you think is a bug. This will allow for more focused reviews.

**Comment by GermanAizek** [comment]
@derekmauro review my new commit and report inaccuracies
https://github.com/abseil/abseil-cpp/commit/e8a89da6b94556143e84cfc834489dbed22b4d84

File: absl/time/internal/cctz/src/time_zone_impl.cc [hunk 1]
```diff
@@ -70,14 +70,13 @@ bool time_zone::Impl::LoadTimeZone(const std::string& name, time_zone* tz) {
...
-  // Load the new time zone (outside the lock).
-  std::unique_ptr<const Impl> new_impl(new Impl(name));
-
   // Add the new time zone to the map.
   std::lock_guard<std::mutex> lock(TimeZoneMutex());
   if (time_zone_map == nullptr) time_zone_map = new TimeZoneImplByName;
   const Impl*& impl = (*time_zone_map)[name];
   if (impl == nullptr) {  // this thread won any load race
+    // Load the new time zone (outside the lock).
+    std::unique_ptr<const Impl> new_impl(new Impl(name));
```
Comment (derekmauro) [inline absl/time/internal/cctz/src/time_zone_impl.cc]: This change is a performance regression. There is no reason to allocate this inside the lock. You've even moved the comment that says "do this outside the lock" into the locked section.

Was this a manual change, or was it generated by an automated tool?

**Review by derekmauro — CHANGES_REQUESTED** [review]

File: absl/time/internal/cctz/src/time_zone_impl.cc (hunk 1 above)
Comment (GermanAizek) [reply absl/time/internal/cctz/src/time_zone_impl.cc]: @derekmauro,
I might not have noticed my fixes, but in this block I did it in vain

Comment (GermanAizek) [reply absl/time/internal/cctz/src/time_zone_impl.cc]: @derekmauro,
fix it https://github.com/abseil/abseil-cpp/commit/926c63e38d30956bbb74c3d34ee54aedb8173c4e

**Review by derekmauro — CHANGES_REQUESTED** [review]
This change doesn't not build with any compiler that I tried. Please ensure the code builds and the tests pass.

**Comment by GermanAizek** [comment]
@derekmauro, how to view test details? in most repositories if you click the 'Details' button.
You are being redirected to an inactive domain http://fusion/

**Comment by derekmauro** [comment]
At the moment you can't, sorry about this. You will have to build it locally.

**Comment by GermanAizek** [comment]
> 
> 
> At the moment you can't, sorry about this. You will have to build it locally.

@derekmauro fix it https://github.com/abseil/abseil-cpp/commit/3b15a2ef282bde28a3e5d82db623cac87711f2e7

**Comment by derekmauro** [comment]
This still does not build on any platform.

**Comment by derekmauro** [comment]
Closing due to lack of progress.

**Pull request closed without merging**